### PR TITLE
Remove time 0.1 as optional dependency

### PR DESCRIPTION
**Pull request by pitdicker**
See https://github.com/chronotope/chrono/issues/1073.

Fixes https://github.com/chronotope/chrono/issues/876, https://github.com/chronotope/chrono/issues/285.

### CODE REVIEW

File: src/date.rs [hunk 1]
```diff
@@ -13,13 +13,13 @@ use core::{fmt, hash};
 #[cfg(feature = "rkyv")]
 use rkyv::{Archive, Deserialize, Serialize};
 
+use crate::duration::Duration as OldDuration;
```
Comment (jtmoon79) [inline src/date.rs:16]: Please add a user-friendly explanation for why `use crate::duration::Duration as OldDuration;` is done here. In other files it's as-is; `use crate::duration::Duration;`. It's confusing that the same type inexplicably changes names in some places.

```rust
// OldDuration is an unfortunate inherited name.
// It does not imply chrono::duration::Duration type should not be used.
// See Issue #1073
use crate::duration::Duration as OldDuration;
```

Obviously, this is a verbose comment to include everywhere there is an `as OldDuration`. Perhaps you can come up with something more concise. Either way, this seemingly random naming oddity should be addressed or it'll be very confusing.

Please do this for all `... as OldDuration;` statements.

**Review by jtmoon79 — CHANGES_REQUESTED** [review]

**Comment by pitdicker** [comment]
I have started a crater(-like) run.

It only checks if a crate still builds with this patch.
If chrono is a dev-dependency or an optional dependency it will not notice a regression.
If a crate fails to build for whatever reason without the patch it will also not be counted.

In the first 500 crates it found 5 regressions, of which as far as I can see 1 is real:
- [ggez](https://crates.io/crates/ggez). Probably spurious.
- [sc-tracing](https://crates.io/crates/sc-tracing). Probably spurious.
- [astrolab-smart-coffee-types](https://crates.io/crates/astrolab-smart-coffee-types). Probably spurious, no dependents.
- [u2f](https://crates.io/crates/u2f). Real. Last updated in 2020, 2 dependents (with no dependents themselves).
- [zellij-server](https://crates.io/crates/zellij-server). Spurious.

**Comment by djc** [comment]
Do you have full error messages for these?

**Comment by pitdicker** [comment]
It is all pretty manual :disappointed:.

This is the error for u2fs-rs (still have the terminal with my manual run open):
```
    Updating git repository `https://github.com/pitdicker/chrono`
    Checking chrono v0.4.26 (https://github.com/pitdicker/chrono?branch=remove_time_0.1#37592e2c)
    Checking u2f v0.2.0 (/home/pitdicker/dev/u2f-rs)
error[E0308]: mismatched types
  --> src/util.rs:25:5
   |
20 | pub fn expiration(timestamp: String) -> Duration {
   |                                         -------- expected `time::Duration` because of return type
...
25 |     now.signed_duration_since(ts.unwrap())
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `time::Duration`, found `chrono::Duration`
   |
   = note: `chrono::Duration` and `time::Duration` have similar names, but are actually distinct types
note: `chrono::Duration` is defined in crate `chrono`
  --> /home/pitdicker/.cargo/git/checkouts/chrono-d04828efa9195fa2/37592e2/src/duration.rs:55:1
   |
55 | pub struct Duration {
   | ^^^^^^^^^^^^^^^^^^^
note: `time::Duration` is defined in crate `time`
  --> /home/pitdicker/.cargo/registry/src/github.com-1ecc6299db9ec823/time-0.1.45/src/duration.rs:45:1
   |
45 | pub struct Duration {
   | ^^^^^^^^^^^^^^^^^^^

warning: use of deprecated method `std::error::Error::description`: use the Display impl or to_string()
  --> src/u2ferror.rs:62:44
   |
62 |             U2fError::OpenSSLError(e) => e.description(),
   |                                            ^^^^^^^^^^^
   |
   = note: `#[warn(deprecated)]` on by default

For more information about this error, try `rustc --explain E0308`.
warning: `u2f` (lib) generated 1 warning
error: could not compile `u2f` due to previous error; 1 warning emitted
```

And astrolab-smart-coffee-types:
```
    Updating git repository `https://github.com/pitdicker/chrono`
    Checking chrono v0.4.26 (https://github.com/pitdicker/chrono?branch=remove_time_0.1#37592e2c)
    Checking actix-http v1.0.1
    Checking bson v0.14.1
error[E0277]: cannot subtract `chrono::Duration` from `Tm`
   --> /home/pitdicker/.cargo/registry/src/github.com-1ecc6299db9ec823/actix-http-1.0.1/src/cookie/jar.rs:224:44
    |
224 |             cookie.set_expires(time::now() - Duration::days(365));
    |                                            ^ no implementation for `Tm - chrono::Duration`
    |
    = help: the trait `Sub<chrono::Duration>` is not implemented for `Tm`
    = help: the following other types implement trait `Sub<Rhs>`:
              <Tm as Sub<time::Duration>>
              <Tm as Sub>

error[E0308]: mismatched types
   --> /home/pitdicker/.cargo/registry/src/github.com-1ecc6299db9ec823/actix-http-1.0.1/src/cookie/mod.rs:736:40
    |
736 |         self.set_expires(time::now() + twenty_years);
    |                                        ^^^^^^^^^^^^ expected `time::Duration`, found `chrono::Duration`
    |
    = note: `chrono::Duration` and `time::Duration` have similar names, but are actually distinct types
note: `chrono::Duration` is defined in crate `chrono`
   --> /home/pitdicker/.cargo/git/checkouts/chrono-d04828efa9195fa2/37592e2/src/duration.rs:55:1
    |
55  | pub struct Duration {
    | ^^^^^^^^^^^^^^^^^^^
note: `time::Duration` is defined in crate `time`
   --> /home/pitdicker/.cargo/registry/src/github.com-1ecc6299db9ec823/time-0.1.45/src/duration.rs:45:1
    |
45  | pub struct Duration {
    | ^^^^^^^^^^^^^^^^^^^

Some errors have detailed explanations: E0277, E0308.
For more information about an error, try `rustc --explain E0277`.
error: could not compile `actix-http` due to 2 previous errors
```

**Comment by jhpratt** [comment]
> It is all pretty manual

Something I've wanted before is to have a crater-like program for ordinary crates. This serves as a good reminder for me to work on it.

Of the two regressions found, one appears unmaintained and the other doesn't appear to actually be used for anything (most days have a download count of zero).

**Comment by pitdicker** [comment]
Now at crate 1000 of the 7227. I will report back when finished.

One more regression found:
[float_duration 0.3.3](https://crates.io/crates/float_duration)

<details>
 <summary>Error</summary>

```
    Updating git repository `https://github.com/pitdicker/chrono`
    Checking chrono v0.4.26 (https://github.com/pitdicker/chrono?branch=remove_time_0.1#37592e2c)
    Checking float_duration v0.3.3 (/home/pitdicker/dev/rust-float-duration)
warning: use of deprecated struct `chrono::Date`: Use `NaiveDate` or `DateTime<Tz>` instead
   --> src/duration.rs:376:50
    |
376 | impl<Tz: chrono::TimeZone> TimePoint for chrono::Date<Tz> {
    |                                                  ^^^^
    |
    = note: `#[warn(deprecated)]` on by default

warning: use of deprecated struct `chrono::Date`: Use `NaiveDate` or `DateTime<Tz>` instead
   --> src/duration.rs:379:50
    |
379 |     fn float_duration_since(self, since: chrono::Date<Tz>) -> Result<FloatDuration, ()> {
    |                                                  ^^^^

error[E0277]: `?` couldn't convert the error to `error::OutOfRangeError`
   --> src/duration.rs:299:71
    |
299 |         let chrono_duration = chrono::Duration::from_std(std_duration)?;
    |                                                                       ^ the trait `From<chrono::OutOfRangeError>` is not implemented for `error::OutOfRangeError`
    |
    = note: the question mark operation (`?`) implicitly performs a conversion on the error value using the `From` trait
    = help: the trait `From<time::OutOfRangeError>` is implemented for `error::OutOfRangeError`
    = note: required for `Result<chrono::Duration, error::OutOfRangeError>` to implement `FromResidual<Result<Infallible, chrono::OutOfRangeError>>`

warning: use of deprecated method `std::error::Error::description`: use the Display impl or to_string()
  --> src/error.rs:25:30
   |
25 |         write!(f, "{}", self.description())
   |                              ^^^^^^^^^^^

For more information about this error, try `rustc --explain E0277`.
warning: `float_duration` (lib) generated 3 warnings
error: could not compile `float_duration` due to previous error; 3 warnings emitted
```
</details>

**Comment by djc** [comment]
@pitdicker awesome work, thanks for getting this done!

**Comment by pitdicker** [comment]
Thank you. But the run is only at 20% now, it wil take some more time...

**Comment by pitdicker** [comment]
I have stopped the run at 3000 crates. What order crates.io returns reverse dependencies I don't know exactly. It is some combination of popularity, or downloads, or number of reverse dependencies. Anyway, after 3000 crates we are in the long tail of one-off projects with less than 1500 downloads, no reverse dependencies, and in quite a number of cases unmaintained.


[exonum-api 1.0.0](https://crates.io/crates/exonum-api). Depends on old version of actix-http-1.0.1, seems to be fixed on master
<details>
 <summary>Error message</summary>

```
    Updating git repository `https://github.com/pitdicker/chrono`
    Checking chrono v0.4.26 (https://github.com/pitdicker/chrono?branch=remove_time_0.1#37592e2c)
    Checking actix-http v1.0.1
error[E0277]: cannot subtract `chrono::Duration` from `Tm`
   --> /home/pitdicker/.cargo/registry/src/github.com-1ecc6299db9ec823/actix-http-1.0.1/src/cookie/jar.rs:224:44
    |
224 |             cookie.set_expires(time::now() - Duration::days(365));
    |                                            ^ no implementation for `Tm - chrono::Duration`
    |
    = help: the trait `Sub<chrono::Duration>` is not implemented for `Tm`
    = help: the following other types implement trait `Sub<Rhs>`:
              <Tm as Sub<time::Duration>>
              <Tm as Sub>

error[E0308]: mismatched types
   --> /home/pitdicker/.cargo/registry/src/github.com-1ecc6299db9ec823/actix-http-1.0.1/src/cookie/mod.rs:736:40
    |
736 |         self.set_expires(time::now() + twenty_years);
    |                                        ^^^^^^^^^^^^ expected `time::Duration`, found `chrono::Duration`
    |
    = note: `chrono::Duration` and `time::Duration` have similar names, but are actually distinct types
note: `chrono::Duration` is defined in crate `chrono`
   --> /home/pitdicker/.cargo/git/checkouts/chrono-d04828efa9195fa2/37592e2/src/duration.rs:55:1
    |
55  | pub struct Duration {
    | ^^^^^^^^^^^^^^^^^^^
note: `time::Duration` is defined in crate `time`
   --> /home/pitdicker/.cargo/registry/src/github.com-1ecc6299db9ec823/time-0.1.45/src/duration.rs:45:1
    |
45  | pub struct Duration {
    | ^^^^^^^^^^^^^^^^^^^

Some errors have detailed explanations: E0277, E0308.
For more information about an error, try `rustc --explain E0277`.
error: could not compile `actix-http` due to 2 previous errors
```
</details>

[nature-db](https://crates.io/crates/nature_db). Depends on old version of actix-http-1.0.1:
<details>
 <summary>Error message</summary>

```
    Updating git repository `https://github.com/pitdicker/chrono`
    Checking chrono v0.4.26 (https://github.com/pitdicker/chrono?branch=remove_time_0.1#37592e2c)
    Checking actix-http v1.0.1
    Checking mysql_common v0.22.2
error[E0277]: cannot subtract `chrono::Duration` from `Tm`
   --> /home/pitdicker/.cargo/registry/src/github.com-1ecc6299db9ec823/actix-http-1.0.1/src/cookie/jar.rs:224:44
    |
224 |             cookie.set_expires(time::now() - Duration::days(365));
    |                                            ^ no implementation for `Tm - chrono::Duration`
    |
    = help: the trait `Sub<chrono::Duration>` is not implemented for `Tm`
    = help: the following other types implement trait `Sub<Rhs>`:
              <Tm as Sub<time::Duration>>
              <Tm as Sub>

error[E0308]: mismatched types
   --> /home/pitdicker/.cargo/registry/src/github.com-1ecc6299db9ec823/actix-http-1.0.1/src/cookie/mod.rs:736:40
    |
736 |         self.set_expires(time::now() + twenty_years);
    |                                        ^^^^^^^^^^^^ expected `time::Duration`, found `chrono::Duration`
    |
    = note: `chrono::Duration` and `time::Duration` have similar names, but are actually distinct types
note: `chrono::Duration` is defined in crate `chrono`
   --> /home/pitdicker/.cargo/git/checkouts/chrono-d04828efa9195fa2/37592e2/src/duration.rs:55:1
    |
55  | pub struct Duration {
    | ^^^^^^^^^^^^^^^^^^^
note: `time::Duration` is defined in crate `time`
   --> /home/pitdicker/.cargo/registry/src/github.com-1ecc6299db9ec823/time-0.1.45/src/duration.rs:45:1
    |
45  | pub struct Duration {
    | ^^^^^^^^^^^^^^^^^^^

    Checking mysql_async v0.23.1
Some errors have detailed explanations: E0277, E0308.
For more information about an error, try `rustc --explain E0277`.
error: could not compile `actix-http` due to 2 previous errors
warning: build failed, waiting for other jobs to finish...
```
</details>


[chrono-utilities 0.0.0-alpha1](https://crates.io/crates/chrono-utilities)
<details>
 <summary>Error message</summary>


```
    Updating git repository `https://github.com/pitdicker/chrono`
    Checking chrono v0.4.26 (https://github.com/pitdicker/chrono?branch=remove_time_0.1#37592e2c)
    Checking chrono-utilities v0.0.0-alpha1 (/home/pitdicker/dev/chrono-utils)
error[E0308]: mismatched types
    --> src/naive/mod.rs:213:33
     |
213  |         self.checked_sub_signed(OldDuration::days(days))
     |              ------------------ ^^^^^^^^^^^^^^^^^^^^^^^ expected `chrono::Duration`, found `oldtime::Duration`
     |              |
     |              arguments to this method are incorrect
     |
     = note: `oldtime::Duration` and `chrono::Duration` have similar names, but are actually distinct types
note: `oldtime::Duration` is defined in crate `time`
    --> /home/pitdicker/.cargo/registry/src/github.com-1ecc6299db9ec823/time-0.1.45/src/duration.rs:45:1
     |
45   | pub struct Duration {
     | ^^^^^^^^^^^^^^^^^^^
note: `chrono::Duration` is defined in crate `chrono`
    --> /home/pitdicker/.cargo/git/checkouts/chrono-d04828efa9195fa2/37592e2/src/duration.rs:55:1
     |
55   | pub struct Duration {
     | ^^^^^^^^^^^^^^^^^^^
note: method defined here
    --> /home/pitdicker/.cargo/git/checkouts/chrono-d04828efa9195fa2/37592e2/src/naive/date.rs:1111:12
     |
1111 |     pub fn checked_sub_signed(self, rhs: OldDuration) -> Option<NaiveDate> {
     |            ^^^^^^^^^^^^^^^^^^

error[E0308]: mismatched types
    --> src/naive/mod.rs:236:33
     |
236  |         self.checked_add_signed(OldDuration::days(days))
     |              ------------------ ^^^^^^^^^^^^^^^^^^^^^^^ expected `chrono::Duration`, found `oldtime::Duration`
     |              |
     |              arguments to this method are incorrect
     |
     = note: `oldtime::Duration` and `chrono::Duration` have similar names, but are actually distinct types
note: `oldtime::Duration` is defined in crate `time`
    --> /home/pitdicker/.cargo/registry/src/github.com-1ecc6299db9ec823/time-0.1.45/src/duration.rs:45:1
     |
45   | pub struct Duration {
     | ^^^^^^^^^^^^^^^^^^^
note: `chrono::Duration` is defined in crate `chrono`
    --> /home/pitdicker/.cargo/git/checkouts/chrono-d04828efa9195fa2/37592e2/src/duration.rs:55:1
     |
55   | pub struct Duration {
     | ^^^^^^^^^^^^^^^^^^^
note: method defined here
    --> /home/pitdicker/.cargo/git/checkouts/chrono-d04828efa9195fa2/37592e2/src/naive/date.rs:1079:12
     |
1079 |     pub fn checked_add_signed(self, rhs: OldDuration) -> Option<NaiveDate> {
     |            ^^^^^^^^^^^^^^^^^^

error[E0277]: cannot subtract `oldtime::Duration` from `NaiveDate`
   --> src/naive/mod.rs:350:49
    |
350 |             Some(week_start) => Some(week_start - OldDuration::days(7)),
    |                                                 ^ no implementation for `NaiveDate - oldtime::Duration`
    |
    = help: the trait `Sub<oldtime::Duration>` is not implemented for `NaiveDate`
    = help: the following other types implement trait `Sub<Rhs>`:
              <NaiveDate as Sub<Days>>
              <NaiveDate as Sub<Months>>
              <NaiveDate as Sub<chrono::Duration>>
              <NaiveDate as Sub>

error[E0277]: cannot subtract `oldtime::Duration` from `NaiveDate`
   --> src/naive/mod.rs:372:49
    |
372 |             Some(week_start) => Some(week_start - OldDuration::days(1)),
    |                                                 ^ no implementation for `NaiveDate - oldtime::Duration`
    |
    = help: the trait `Sub<oldtime::Duration>` is not implemented for `NaiveDate`
    = help: the following other types implement trait `Sub<Rhs>`:
              <NaiveDate as Sub<Days>>
              <NaiveDate as Sub<Months>>
              <NaiveDate as Sub<chrono::Duration>>
              <NaiveDate as Sub>

error[E0277]: cannot add `oldtime::Duration` to `NaiveDate`
   --> src/naive/mod.rs:486:49
    |
486 |             Some(week_start) => Some(week_start + OldDuration::days(7)),
    |                                                 ^ no implementation for `NaiveDate + oldtime::Duration`
    |
    = help: the trait `Add<oldtime::Duration>` is not implemented for `NaiveDate`
    = help: the following other types implement trait `Add<Rhs>`:
              <NaiveDate as Add<Days>>
              <NaiveDate as Add<Months>>
              <NaiveDate as Add<chrono::Duration>>

error[E0277]: cannot add `oldtime::Duration` to `NaiveDate`
   --> src/naive/mod.rs:508:49
    |
508 |             Some(week_start) => Some(week_start + OldDuration::days(6)),
    |                                                 ^ no implementation for `NaiveDate + oldtime::Duration`
    |
    = help: the trait `Add<oldtime::Duration>` is not implemented for `NaiveDate`
    = help: the following other types implement trait `Add<Rhs>`:
              <NaiveDate as Add<Days>>
              <NaiveDate as Add<Months>>
              <NaiveDate as Add<chrono::Duration>>

Some errors have detailed explanations: E0277, E0308.
For more information about an error, try `rustc --explain E0277`.
error: could not compile `chrono-utilities` due to 6 previous errors
```
</details>

[game_time 0.2.0](https://crates.io/crates/game_time) => depends on float_duration


[eternal 0.3.2](https://crates.io/crates/eternal). github repo no longer exists.

<details>
 <summary>Error message</summary>

```
    Updating git repository `https://github.com/pitdicker/chrono`
    Checking chrono v0.4.26 (https://github.com/pitdicker/chrono?branch=remove_time_0.1#37592e2c)
    Checking eternal v0.3.2 (/home/pitdicker/dev/eternal-0.3.2)
error[E0277]: cannot subtract `chrono::Duration` from `Tm`
   --> src/http/cookie/jar.rs:224:44
    |
224 |             cookie.set_expires(time::now() - Duration::days(365));
    |                                            ^ no implementation for `Tm - chrono::Duration`
    |
    = help: the trait `Sub<chrono::Duration>` is not implemented for `Tm`
    = help: the following other types implement trait `Sub<Rhs>`:
              <Tm as Sub<time::Duration>>
              <Tm as Sub>

error[E0308]: mismatched types
   --> src/http/cookie/mod.rs:736:40
    |
736 |         self.set_expires(time::now() + twenty_years);
    |                                        ^^^^^^^^^^^^ expected `time::Duration`, found `chrono::Duration`
    |
    = note: `chrono::Duration` and `time::Duration` have similar names, but are actually distinct types
note: `chrono::Duration` is defined in crate `chrono`
   --> /home/pitdicker/.cargo/git/checkouts/chrono-d04828efa9195fa2/37592e2/src/duration.rs:55:1
    |
55  | pub struct Duration {
    | ^^^^^^^^^^^^^^^^^^^
note: `time::Duration` is defined in crate `time`
   --> /home/pitdicker/.cargo/registry/src/github.com-1ecc6299db9ec823/time-0.1.45/src/duration.rs:45:1
    |
45  | pub struct Duration {
    | ^^^^^^^^^^^^^^^^^^^

Some errors have detailed explanations: E0277, E0308.
For more information about an error, try `rustc --explain E0277`.
error: could not compile `eternal` due to 2 previous errors
```
</details>

----

This brings the number my primitive crater run found to 9 crates. 3 of those depend fail to compile the old dependency actix-http-1.0.1. And eternal 0.3.2 seems to have vendored that code.

All haven't seen a release in a couple of years, and don't really effect the crates.io ecosystem.

**Comment by djc** [comment]
So to put everything together:

* [astrolab-smart-coffee-types](https://crates.io/crates/astrolab-smart-coffee-types)
  * No dependents
  * Some download spikes, but 0 downloads on most days
  * Last release 2.5 years ago
  * No public repository
* [u2f](https://crates.io/crates/u2f)
  * 2 dependents, with a total of 219 all-time downloads
    * [tauri-plugin-authenticator](https://crates.io/crates/tauri-plugin-authenticator) was released 1 day ago by the Tauri project
  * ~10 daily downloads with some spikes in the hundreds over the past 2 weeks (from Tauri folks?)
  * Last release 3.5 years ago
  * Last commit 3.5 years ago
* [float_duration](https://crates.io/crates/float_duration)
  * 2 dependents, with a total of ~3k all-time downloads
    * [osmon](https://crates.io/crates/osmon) had its most recent release 1 month ago, but it's getting 0 downloads on most days with some single-digit spikes
  * Last release 6 years ago
  * Last commit 4 years ago
* [exonum-api 1.0.0](https://crates.io/crates/exonum-api) -> via actix-http 1.0.1
  * 6 dependents, all from the Exonum project, -testkit has 16k all-time downloads
  * ~5 daily downloads
  * Last release 3 years ago
  * Last commit is a few months old, last non-Dependabot commit is from October
* [nature-db](https://crates.io/crates/nature_db) -> via actix-http 1.0.1
  * 1 dependent, from the same author/project, 3k all-time downloads
  * ~1 daily download
  * Last release 3 years ago
  * Last commit 3 years ago, repo is archived
* [chrono-utilities 0.0.0-alpha1](https://crates.io/crates/chrono-utilities)
  * No dependents
  * ~7 daily downloads
  * Last release 3 years ago
  * Last commit 3 years ago
* [game_time 0.2.0](https://crates.io/crates/game_time) => depends on float_duration
  * Same author as float_duration
* [eternal 0.3.2](https://crates.io/crates/eternal)
  * 2 dependents related to the same project, 2k all-time downloads
  * ~1 daily download
  * Last release 3 years ago
  * No public repository

I've asked an Actix maintainer whether it would be feasible to release an actix-http 1.0.2 to fix those dependencies. We should probably do some outreach for u2f/Tauri and float_duration/Osmon.

**Comment by pitdicker** [comment]
Made a PR to u2f: https://github.com/wisespace-io/u2f-rs/pull/9

**Comment by djc** [comment]
Filed a heads-up against the tauri-plugin-authenticator repo in https://github.com/tauri-apps/plugins-workspace/issues/411. Seems they don't want us to block on their usage.

**Comment by pitdicker** [comment]
@djc What are your thoughts on making the jump?

**Comment by pitdicker** [comment]
There hasn't been any activity on exonum.

File: src/date.rs (hunk 1 above)
Comment (jtmoon79) [reply src/date.rs:16]: It should be addressed in a code comment. As-is, having `crate::duration::Duration` treated differently in different files is confusing.

Comment (pitdicker) [reply src/date.rs]: I added a short comment to the imports that this is to avoid confusion with `Duration` from the standard library in the docs.

**Review by jtmoon79 — CHANGES_REQUESTED** [review]
This still jumps between two references for `crate::duration::Duration`

    use crate::duration::Duration as OldDuration;

and

    use crate::duration::Duration;

Pick one and be consistent about it everywhere. If reference `Duration` causes name collisions then I suggest `ChronoDuration`.

**Comment by pitdicker** [comment]
@jtmoon79 I want to point out that the imports here and in https://github.com/chronotope/chrono/pull/1137 are pre-existing, and changing them was not the goal of this PR. If it is worth changing I think that should be a separate PR.

**Comment by jtmoon79** [comment]
> @jtmoon79 I want to point out that the imports here and in #1137 are pre-existing, and changing them was not the goal of this PR. If it is worth changing I think that should be a separate PR.

Okay.

**Review by jtmoon79 — APPROVED** [review]
LGTM

The sometimes aliasing of `chrono::duration::Duration as OldDuration` can be changed in another PR.

**Comment by djc** [comment]
Given that we have so many changes on 0.4.x now, I feel uncomfortable putting this out into the same release as the other changes. Let's do another release for all the other stuff, wait a few days for feedback and then merge and release this.

**Comment by pitdicker** [comment]
Made a list of changes since the last release: https://gist.github.com/pitdicker/e13d4295cca6d5c118b88d621ce1e2d8

**Comment by djc** [comment]
Can you make that into a draft GitHub release, and make sure to call out external contributors (anyone not you or me) as we've done in previous release notes?

**Comment by djc** [comment]
Let's get #1145 and #1225 merged before we release.

**Comment by pitdicker** [comment]
> Let's do another release for all the other stuff, wait a few days for feedback and then merge and release this.

Are we ready to merge? Not that I have been looking forward to it or something :laughing:

**Review by djc — APPROVED** [review]

**Comment by djc** [comment]
Yes, let's do it! Can you submit a version bump PR and draft release notes?

**Comment by detro** [comment]
I just want to say: thank you so much for taking the time to make this happen.

Sometimes OS development is less about "solving great problems in great way", and more about supporting your community and a bit of grinding.

Everyone appreciates this effort: I'm confident I can say this for all rust developers depending on crono.

**Comment by djc** [comment]
Well, we'll see how much hidden breakage comes out of the woodwork and/or how many people will complain that we're breaking the semver contract.

**Pull request merged**